app/assets/javascripts/application/multiple-files.js

```diff
@@ -0,0 +1,211 @@
+mumuki.load(function () {
+
+  function File(tab, editor) {
+    this.tab = tab;
+    this.editor = editor;
+  }
+
+  File.FILE_NAME_CLASS = '.file-name';
+  File.DELETE_BUTTON_CLASS = '.delete-file-button';
+
+  File.prototype = {
+    initialize: function(name) {
+      this.setName(name);
+      this.unselect();
+
+      return this;
+    },
+
+    getName: function() {
+      return this.tab.find(File.FILE_NAME_CLASS).text();
+    },
+
+    setName: function(name) {
+      return this.tab.find(File.FILE_NAME_CLASS).text(name);
+    },
+
+    isSelected: function() {
+      return this.tab.hasClass("active");
+    },
+
+    setUpOnRemove: function(handler) {
+      this.tab.find(File.DELETE_BUTTON_CLASS).click(function() {
+        handler(this);
+      }.bind(this));
+    },
+
+    remove: function() {
+      var wasSelected = this.isSelected();
+
+      this.tab.remove();
+      this.editor.remove();
+
+      return wasSelected;
+    },
+
+    select: function() {
+      this._selectElement(this.tab);
+      this._selectElement(this.editor);
+    },
+
+    unselect: function() {
+      this._unselectElement(this.tab);
+      this._unselectElement(this.editor);
+    },
+
+    _selectElement: function(element) {
+      element.addClass('active');
+      element.addClass('in');
+    },
+
+    _unselectElement: function(element) {
+      element.removeClass('active');
+      element.removeClass('in');
+    }
+  };
+
+  function FileControls(tabsContainer, editorsContainer) {
+    this.tabsContainer = tabsContainer;
+    this.editorsContainer = editorsContainer;
+
+    this.MAX_TABS = 5;
+
+    this._addFileButton = this.tabsContainer.siblings('.add-file-button');
+    this._updateButtonsVisibility();
+  }
+
+  FileControls.prototype = {
+    files: function() {
+      var editors = this.editors();
+
+      return this.tabs().map(function(i, tab) {
+        return new File($(tab), $(editors[i]));
+      });
+    },
+
+    tabs: function() {
+      return this.tabsContainer.children();
+    },
+
+    editors: function() {
+      return this.editorsContainer.find('.file-editor')
+    },
+
+    setUpAddFile: function() {
+      this._addFileButton.click(function() {
+        this._addFile();
+      }.bind(this));
+    },
+
+    setUpDeleteFiles: function() {
+      this.files().each(function(i, file) {
+        this.setUpDeleteFile(file);
+      }.bind(this));
+    },
+
+    setUpDeleteFile: function(file) {
+      file.setUpOnRemove(this._deleteFile.bind(this));
+    },
+
+    _addFile: function() {
+      var name = prompt('Insert a file name'); // TODO: i18n, or improve somehow
```

Comment:
Oops! @fedescarpa what can we do here? It is very important that the UI is fully internationalized